### ISSUE TITLE
Huge numbers of files in ~/.lpass/upload-queue

### ISSUE THREAD

**Issue by davidblewett** [lastpass/lastpass-cli#81]
I've somehow managed to accumulate almost 1M files in `~/.lpass/upload-queue`, occupying almost 4GB of disk space. Is it safe to nuke the contents of this directory? I noticed because the process that is supposed to handle this queue was consuming a constant amount of CPU (not 100% though; 30-40%).

I'm currently using binaries built from version 6282e79 (0.5.0).

**Comment by davidblewett** [comment]
It appears something in that commit is not interacting well with my use case ( see https://gist.github.com/davidblewett/53047c4c7757b663c11b ). When I run `lp-ssh-add.sh` now, the script hangs on L23 for the very first key and is unkillable with Ctl-C. I have to kill it from another shell. As long as the script is running, it's generating thousands of files in `~/.lpass/upload-queue`. Why is just dumping the contents of different keys triggering an upload of anything?

**Comment by bcopeland** [comment]
Hmm, not sure yet - but note output of show has changed for secure notes, you don't need to use grep anymore because these notes are parsed and you can extract them with --field.

**Comment by bcopeland** [comment]
Regarding output differences:
 * Notes field will now have an extra newline
 * Passphrase will now have an extra leading space

I think what's happening is the latter causes ssh-askpass to fail (provide wrong passphrase) and so ssh-add gets stuck in a loop asking for it over and over again.  See my updated gist which may fix things for you (although you also might want to come up with a way to abort the process if lp-ssh-askpass is failing).

upload-queue gets used to log accesses for history/auditing purposes (see cmd-show.c:226).

**Comment by davidblewett** [comment]
I think I've solved my issues in the gist; still surprised it's possible to generate gigabytes of this audit log.

**Comment by bcopeland** [comment]
Yes, perhaps we should rate limit it somehow - once it gets over some threshold just error out so one doesn't get stuck sending us gigabytes of updates.

**Comment by glyph** [comment]
I also ran into this problem, after changing my master password.  No `ssh-agent` loop in my case, I just tried to create a new item with `lpass generate` and `lpass info`, and it happily generated gigs of `upload-queue` on its own.  Sadly I deleted the `~/.lpass` directory so I don't have a copy of the buggy state, but it does seem to be working fine now.

**Comment by wkoszek** [comment]
@bcopeland Looks like this issue is solved. Can we close the issue?

**Comment by cizmazia** [comment]
Possibly related #265
